### ISSUE TITLE
`httplib2` to `requests` migration

### ISSUE THREAD

**Issue by mchavda-splunk** [splunk/addonfactory-ta-library-python#50]
Currently, `splunktalib` uses the `httplib2` internally to make api call. In the next release of `splunktalib` (3.0.0), we will be replacing the `httplib2` library with the `requests` library. This issue provides the change details.

## Difference between `httplib2` and `requests`:
1. httplib2 returns ```(response_headers, content)``` and requests returns `response`
2. In httplib2, the ```content``` is a type of bytes object which needs to be converted into a str object. In requests, we can directly get the str object using the ```response.text```

Example:

Using `httplib2`:
```python
import httplib2
h = httplib2.Http()
(response_headers, content) = h.request("http://example.org/", "GET")
print(response_headers.status)
data = content.decode('utf-8')
```

Using `requests`:
```python
import requests
response = requests.get("GET", "http://example.org/")
print(response.status_code)
data = response.text
```

## Code change require to use the splunktalib v3.0.0
**Note**: If you are using the ```splunktalib.rest``` directly in your code then you need to make below changes in your source code to use the splunktalib v3.0.0.

1. ```splunktalib.rest.build_http_connection```
* The ```build_http_connection``` function is removed in the splunktalib v3.0.0. We recommend users to directly use the ```requests.request``` function to make the API calls. Doc: https://docs.python-requests.org/en/latest/api/

2. ```splunktalib.rest.splunkd_request``` and ```splunktalib.rest.code_to_msg```
* The output signature of ```splunkd_request``` function is changed as per the requests library standards.
* The input signature of  ```code_to_msg``` function is changed as per the requests library standards.

Before:
```python
response_headers, content = splunkd_request(<params>)
print(response_headers.status)
data = content.decode('utf-8')

msg = code_to_msg(response_headers, content)
```

After:
```python
response = splunkd_request(<params>)
print(response.status_code)
data = response.text

msg = code_to_msg(response)
```

**Comment by arys-splunk** [comment]
This is already available in beta version of `splunktalib`: https://pypi.org/project/splunktalib/3.0.0b1/

**Comment by arys-splunk** [comment]
This is released as 3.0.0 release (https://github.com/splunk/addonfactory-ta-library-python/releases/tag/v3.0.0).
